### ISSUE TITLE
Adaptive lighting feature unable to connect 

### ISSUE THREAD

**Issue by skir0cket** [bwp91/homebridge-govee#28]
Hi im using the Govee H1682 tv leds they are working fine with the plug-in but I’m unable to connect them in home to use the adaptive feature I get a cannot communicate with bridge accessorie error. I’m using the v2.51 plugin and latest beta of homebridge. 

`DEBUG:
[%s] Encountered error on CharacteristicValueTransitionControl characteristic: TypeError: Cannot read property 'readUInt16LE' of undefined
    at AdaptiveLightingController.handleTransitionControlUpdateTransition (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/controller/AdaptiveLightingController.ts:1106:103)
    at AdaptiveLightingController.handleTransitionControlWrite (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/controller/AdaptiveLightingController.ts:1027:45)
    at CharacteristicValueTransitionControl.setHandler (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/controller/AdaptiveLightingController.ts:890:23)
    at CharacteristicValueTransitionControl.<anonymous> (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1483:42)
    at step (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:140:27)
    at Object.next (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:121:57)
    at /usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:114:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:110:16)
    at CharacteristicValueTransitionControl.Characteristic.handleSetRequest (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/dist/lib/Characteristic.js:718:24)`

**Comment by bwp91** [comment]
Hi @skir0cket 
Please can you post the complete error message from the Homebridge log? And when does this message appear? When you try to turn on adaptive lighting? when it periodically tries to send a new colour?
Also, I know you stated you are using the latest beta of Homebridge, but please double check the version you are using is beta-40.
Thanks

**Comment by skir0cket** [comment]
Hi yes its beta 40 I have added the log above, I get a popup in home that says adaptive lighting is now available when I click to set up lights I get the error when trying to connect, all other features work when controlling them

**Comment by bwp91** [comment]
Ignoring the popup, what happens if you go to the device icon in home app, open it, and click the adaptive lighting colour icon?

**Comment by skir0cket** [comment]
it doesn't switch over

**Comment by bwp91** [comment]
Does a log message appear when you click the icon?

**Comment by skir0cket** [comment]
` Encountered error on CharacteristicValueTransitionControl characteristic: TypeError: Cannot read property 'readUInt16LE' of undefined
    at AdaptiveLightingController.handleTransitionControlUpdateTransition (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/controller/AdaptiveLightingController.ts:1106:103)
    at AdaptiveLightingController.handleTransitionControlWrite (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/controller/AdaptiveLightingController.ts:1027:45)
    at CharacteristicValueTransitionControl.setHandler (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/controller/AdaptiveLightingController.ts:890:23)
    at CharacteristicValueTransitionControl.<anonymous> (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1483:42)
    at step (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:140:27)
    at Object.next (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:121:57)
    at /usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:114:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:110:16)
    at CharacteristicValueTransitionControl.Characteristic.handleSetRequest (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/dist/lib/Characteristic.js:718:24)`

**Comment by bwp91** [comment]
Please try deleting the file (the ### will be series of numbers/letters)
```
/var/lib/homebridge/persist/ControllerStorage.###########.json
```
and restart Homebridge to see if this helps?

**Comment by skir0cket** [comment]
Hi that file does not exist in that directory

**Comment by bwp91** [comment]
ok have you tried restarted homebridge anyway?

**Comment by skir0cket** [comment]
yes I have restarted and also re added the plugin

**Comment by skir0cket** [comment]
there is a bunch of numbers and letters above the log not sure if this will give anymore info as to the issue, sorry just noticed.

`%s] DEBUG: 'Av8B/wEBDQImARDE8z2pY8lMD4uJbDajPu+eAgjYLzwFkwAAAAMIiQdDETA8S4wDAQEF/wEPAQRVVdW/AgRV1bBDAwEAAAABEgEEbMHWvwIEjmOwQwMEaOgOAAAAARIBBIMt2L8CBMdxr0MDBEB3GwAAAAESAQSDLdi/AgTH8a5DAwRAdxsAAAABGAEEbMHWvwIEjuOuQwMEQHcbAAQEgO42AAAAARIBBIMt2L8CBMfxrkMDBEB3GwAAAAEYAQQ/6dO/AgQcR69DAwRAdxsABASA7jYAAAABEgEE5DjOvwIEOQ6wQwMEQHcbAAAAARIBBHIcx78CBBzHsUMDBEB3GwAC/wAAAf8BEgEEvLu7vwIEVVWzQwMEQHcbAAAAARIBBAZbsL8CBI5jtkMDBEB3GwAAAAESAQQF/1D6pL8CBMfxuUMDBEB3GwAAAAESAQTe3Z2/AgSrKr9DAwRAdxsAAAABEgEE9UmfvwIE5LjGQwMEQHcbAAAAARIBBH3Sp78CBDkO0EMDBEB3GwAAAAESAQSlT7q/AgQcx9tDAwRAdxsAAAABEgEEERHRvwIEq6rnQwMEQHcbAAAAARIBBKuq6r8CBKuq80MDBEB3GwAAAAESAQQAAADAAgQAAP5DAwRAdxsAAAABEgEEW7AFwAIE5DgCRAMEQHcbAAAAARIBBCIiAgL/wAIEVQH/FQJEAwRAdxsAAAABEgEE0if9vwIEx/EBRAMEQHcbAAAAARIBBAZb8L8CBMfxAEQDBEAF/3cbAAAAARIBBNiC7b8CBI6jAUQDBEB3GwAAAAEYAQR90ue/AgQcBwBEAwRAdxsABASAy6QAAAABEgEEC7bgvwIEHMf7QwMEQHcbAAAAARIBBFVV1b8CBFVV9UMDBEB3GwAAAAESAQSf9Mm/AgSOY+5DAwRAdxsAAAABEgEE0ie9vwIEjmPnQwMEQHcbAAAAARIBBNiCrb8CBBzH30MDBEB3GwAAAAESAQT1SZ+/AgTkuNhDAwRAdxsAAAABEgEEP+mTvwIEAtUcx9FDAwQBzUB3GwAAAAESAQQREZG/AgSrqstDAwRAdxsAAAABEgEEgy2YvwIEx/HFQwMEQHcbAAAAARIFkQEEZmamvwIEAIDAQwMEQHcbAAAAARIBBGELtr8CBHKcu0MDBEB3GwAAAAESAQRERMS/AgSrqrdDAwRAdxsAAAABEgEEzczMvwIEAAC0QwMEQHcbAAAAARIBBD/p078CBBzHsUMDBEB3GwAAAAESAQRVVdW/AgRV1bBDAwTYjgwAAgEKAwwBBAoAAAACBGQAAAAIBOCTBAA='`

**Comment by bauer-andreas** [comment]
@skir0cket those "number and letters" is the important stuff. This is basically the data for the transition schedule. I have a look at it. This is probably something we haven't yet discovered about Adaptive Lighting.

**Comment by skir0cket** [comment]
ok thanks for your time and help.

**Comment by bauer-andreas** [comment]
Identified the issue. Fixed with homebridge 1.3.0 Beta 41.
Thanks for reporting 👏 

Issue can be closed.

**Comment by skir0cket** [comment]
thanks its working great now!
